pkg/apis/application/v1alpha1/types.go

```diff
@@ -690,15 +690,16 @@ func (o SyncOptions) HasOption(option string) bool {
 }
 
 //GetNSMetaData returns match json string if the list of sync options contains given sync option
-func (o SyncOptions) GetNSMetaData(option string) (m string) {
+func (o SyncOptions) GetNSMetaData(option string) map[string]interface{} {
 
+	r := make(map[string]interface{})
 	for _, i := range o {
 		s := strings.Split(i, "=")
```

Comment:
Could you do `strings.SplitN(i, "=", 2)` and add a test to make sure we allow annotations that contain `=` signs?